metadata-ingestion/src/datahub/ingestion/source/sql/mssql/source.py

```diff
@@ -348,7 +348,151 @@ def get_schema_level_workunits(
                     f"Failed to list jobs due to error {e}",
                 )
 
+    def _detect_rds_environment(self, conn: Connection) -> bool:
+        """
+        Detect if we're running in an RDS/managed environment vs on-premises.
+        Returns True if RDS/managed, False if on-premises.
+        """
+        try:
+            # Try to access system tables directly - this typically fails in RDS
+            conn.execute("SELECT TOP 1 * FROM msdb.dbo.sysjobs")
+            logger.debug(
+                "Direct table access successful - likely on-premises environment"
+            )
+            return False
+        except Exception:
+            logger.debug("Direct table access failed - likely RDS/managed environment")
+            return True
+
     def _get_jobs(self, conn: Connection, db_name: str) -> Dict[str, Dict[str, Any]]:
+        """
+        Get job information with environment detection to choose optimal method first.
+        """
+        jobs: Dict[str, Dict[str, Any]] = {}
+
+        # Detect environment to choose optimal method first
+        is_rds = self._detect_rds_environment(conn)
+
+        if is_rds:
+            # RDS/Managed environment - try stored procedures first
+            try:
+                jobs = self._get_jobs_via_stored_procedures(conn, db_name)
+                logger.info(
+                    "Successfully retrieved jobs using stored procedures (RDS/managed environment)"
+                )
+                return jobs
+            except Exception as sp_error:
+                logger.warning(
+                    f"Failed to retrieve jobs via stored procedures in RDS environment: {sp_error}"
+                )
+                # Try direct query as fallback (might work in some managed environments)
+                try:
+                    jobs = self._get_jobs_via_direct_query(conn, db_name)
+                    logger.info(
+                        "Successfully retrieved jobs using direct query fallback in RDS environment"
+                    )
+                    return jobs
+                except Exception as direct_error:
+                    logger.error("Both methods failed in RDS environment")
+                    self.report.report_failure(
+                        "jobs",
+                        f"Failed to retrieve jobs in RDS environment. "
+                        f"Stored procedure error: {sp_error}. Direct query error: {direct_error}",
+                    )
```

Comment:
Given the title has high visibility in the UI, we should set a more descriptive error report title than just `"jobs"`, _eg_ `Get jobs error` or something like that.

Same comment applies to others `report_(warning|error)` in this PR.